btree/btree.c

```diff
@@ -4466,12 +4492,425 @@ static void m0_btree_ut_multi_stream_kv_oper(void)
+	uint64_t               value[ti->ti_value_size/sizeof(uint64_t)];
+	m0_bcount_t            ksize         = sizeof key;
+	m0_bcount_t            vsize         = sizeof value;
+	void                  *k_ptr         = &key;
+	void                  *v_ptr         = &value;
+	struct m0_btree_rec    rec;
+	struct m0_btree_cb     ut_cb;
+	struct cb_data         data;
+
+	uint64_t               get_key[ti->ti_key_size/sizeof(uint64_t)];
+	uint64_t               get_value[ti->ti_value_size / sizeof(uint64_t)];
+	m0_bcount_t            get_ksize     = sizeof get_key;
+	m0_bcount_t            get_vsize     = sizeof get_value;
+	void                  *get_k_ptr     = &get_key;
+	void                  *get_v_ptr     = &get_value;
+	struct m0_btree_rec    get_rec;
+	struct m0_btree_cb     ut_get_cb;
+	struct cb_data         get_data;
+
+	uint64_t               key_iter;
+	struct m0_btree_op     kv_op     = {};
+	struct m0_btree       *tree;
+	struct m0_be_tx       *tx        = NULL;
+
+	/**
+	 *  Currently our thread routine only supports Keys and Values which are
+	 *  a multiple of 8 bytes.
+	 */
+	M0_ASSERT(ti->ti_key_size % sizeof(uint64_t) == 0);
+	M0_ASSERT(ti->ti_value_size % sizeof(uint64_t) == 0);
+
+	/** Prepare transaction to capture tree operations. */
+	m0_be_tx_init(tx, 0, NULL, NULL, NULL, NULL, NULL, NULL);
+	m0_be_tx_prep(tx, NULL);
+
+	key_iter = ti->ti_key_first;
+
+	rec.r_key.k_data   = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize);
+	rec.r_val          = M0_BUFVEC_INIT_BUF(&v_ptr, &vsize);
+
+	data.key           = &rec.r_key;
+	data.value         = &rec.r_val;
+
+	ut_cb.c_act        = btree_kv_put_cb;
+	ut_cb.c_datum      = &data;
+
+	get_rec.r_key.k_data   = M0_BUFVEC_INIT_BUF(&get_k_ptr, &get_ksize);
+	get_rec.r_val          = M0_BUFVEC_INIT_BUF(&get_v_ptr, &get_vsize);
+
+	get_data.key           = &get_rec.r_key;
+	get_data.value         = &get_rec.r_val;
+	get_data.check_value   = true;
+
+	ut_get_cb.c_act        = btree_kv_get_cb;
+	ut_get_cb.c_datum      = &get_data;
+
+	tree                   = ti->ti_tree;
+
+	/** Wait till all the threads have been initialised. */
+	while (!thread_start)
+	;
+
+	while (key_iter <= ti->ti_key_last) {
+		uint64_t  key_start;
+		uint64_t  key_end;
+		uint64_t  keys_put_count = 0;
+		uint64_t  keys_found_count = 0;
+		int       i;
+		int32_t   r;
+		uint64_t  iter_dir;
+		uint64_t  del_key;
+
+		key_start = key_iter;
+		if (ti->ti_random_bursts) {
+			random_r(&ti->ti_random_buf, &r);
+			key_end = (r % (ti->ti_key_last - key_start)) +
+				  key_start;
+		} else
+			key_end = ti->ti_key_last;
+
+		/** PUT keys and their corresponding values in the tree. */
+		while (key_start <= key_end) {
+			/**
+			 *  Embed the thread-id in LSB so that different threads
+			 *  will target the same node thus causing race
+			 *  conditions useful to mimic and test btree operations
+			 *  in a loaded system.
+			 */
+			key[0] = (key_start << (sizeof(ti->ti_thread_id) * 8)) +
```

Comment:
`sizeof m0_bitmap_set_nr` is preferrable.

Reply:
I cannot use m0_bitmap_set_nr() since ti_thread_id is not of type bitmap.